pkg/prometheus/promcfg.go

```diff
@@ -2281,6 +2282,173 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			Value: configs,
 		})
 	}
+
+	//ConsulSDConfig
+	if len(sc.Spec.ConsulSDConfigs) > 0 {
```

Comment:
The generateScrapeConfig() method starts being quite long. As a follow-up, it would be nice to break it up in smaller pieces.